tidy3d/components/source.py

```diff
@@ -105,6 +105,8 @@ def spectrum(
         stop_ind = relevant_time_inds[0][-1]
         time_amps = time_amps[start_ind:stop_ind]
         times_cut = times[start_ind:stop_ind]
+        # TODO: this fails if tim_amps is all 0 (for example if amplitude=0.0)
+        # maybe if this case is detected, we automatically return `np.zeros_like(freqs)`?
```

Comment:
Sounds like a good solution, just add it here so it doesn't fall through the cracks?

Reply:
ok added